Bio/SeqIO/QualityIO.py

```diff
@@ -1242,24 +1269,28 @@ def FastqSolexaIterator(source, alphabet=None, title2ids=None):
 
     for title_line, seq_string, quality_string in FastqGeneralIterator(source):
         if title2ids:
-            id, name, descr = title_line
+            id, name, descr = title2ids(title_line)
```

Comment:
This is a bug that was found due to type checking :-)

Reply:
Ah - specific to the legacy Solexa parser, and a rarely used option - probably why it was missed. As an aside, in hindsight I regret copying the ``title2ids`` behaviour from the older FASTA parser.

Would you might putting that one line fix in as a separate pull request? Or may I, with an acknowledgement?

Reply:
https://github.com/biopython/biopython/pull/3175 :-)

Reply:
Thank you.